src/Microsoft.Azure.WebJobs.Host/Configuration/CoreWebJobsOptionsSetup.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Host.Configuration
+{
+    internal class CoreWebJobsOptionsSetup<TOptions> : IConfigureOptions<TOptions> where TOptions : class
```

Comment:
I debated just pushing the pattern with the delegate, but this goal of this type is actually to simplify that and do exactly what you're describing; avoid additional types to configure options. 

This is now a single type that handles binding against the root configuration, abstracting away the logic to grab the correct root scope and binding. This will avoid a lot of repetitive code as we properly configure all of the different core options types.